Bottom: 6fb26f02c06c27cb2b94fd21d21273260c3c4563
Top:    3e3c89beff00570aab96ab4c0bd6a397aaae83a4
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-17 12:42:17 -0700

Refresh of fork

---

diff --git a/src/preload.c b/src/preload.c
index cdf1ea6..00ca24e 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -460,7 +460,6 @@ int accept(int socket, struct sockaddr *addr, socklen_t *addrlen)
 
 static int connect_fork(int socket, const struct sockaddr *addr, socklen_t addrlen)
 {
-	struct sockaddr_in *sin;
 	int fd, ret;
 	uint32_t msg;
 
@@ -505,6 +504,9 @@ int connect(int socket, const struct sockaddr *addr, socklen_t addrlen)
 
 		rclose(fd);
 		fd = ret;
+		break;
+	default:
+		break;
 	}
 
 	return real.connect(fd, addr, addrlen);
@@ -821,7 +823,8 @@ pid_t fork(void)
 	struct sockaddr_in6 sin6;
 	pid_t pid;
 	sem_t *sem;
-	int lfd, sfd, dfd, ret, len, param;
+	int lfd, sfd, dfd, ret, param;
+	socklen_t len;
 	uint32_t msg;
 
 	init_preload();
@@ -831,7 +834,7 @@ pid_t fork(void)
 		goto out;
 
 	len = sizeof sa;
-	ret = real.getsockname(sfd, &sa, &len);
+	ret = real.getsockname(sfd, (struct sockaddr *) &sa, &len);
 	if (ret)
 		goto out;
 	sin6.sin6_family = sa.ss_family;
